src/uct/base/uct_md.c

```diff
@@ -308,8 +312,10 @@ ucs_status_t uct_iface_open(uct_md_h md, uct_worker_h worker,
                             uct_iface_h *iface_p)
 {
     uct_tl_component_t *tlc;
+    uct_md_attr_t md_attr;
 
-    tlc = uct_find_tl_on_md(md->component, params->mode.device.tl_name);
+    md->ops->query(md, &md_attr);
```

Comment:
better use uct_md_query(), and also need to check its return status.

Reply:
but uct_md_query() does more than only query:
```
    /* MD component name + data */
    memcpy(md_attr->component_name, md->component->name, UCT_MD_COMPONENT_NAME_MAX);
    md_attr->rkey_packed_size += UCT_MD_COMPONENT_NAME_MAX;
```

Reply:
still, better use API function that to call function pointer directly

Reply:
but it will increase the md_attr->rkey_packed_size value to a wrong value.

Reply:
it's not the wrong value, it is correct buffer size which should be passed to uct_rkey_pack().
also, i don't see that rkey pack is used in this PR..